app/src/androidTest/java/org/mozilla/fenix/ui/SettingsPrivacyTest.kt

```diff
@@ -194,6 +199,98 @@ class SettingsPrivacyTest {
         }
     }
 
+    @Test
+    fun verifyPrivateBrowsingMenuItemsTest() {
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openPrivateBrowsingSubMenu {
+            verifyAddPrivateBrowsingShortcutButton()
+            verifyOpenLinksInPrivateTab()
+            verifyOpenLinksInPrivateTabOff()
+        }.goBack {
+            verifySettingsView()
+        }
+    }
+
+    @Test
+    fun openExternalLinksInPrivateTest() {
+        setOpenLinksInPrivateOn()
+
+        openAppFromExternalLink("https://google.com")
+
+        browserScreen {
+        }.openHomeScreen {
+            verifyPrivateSessionHeader()
+        }
+
+        setOpenLinksInPrivateOff()
+
+        openAppFromExternalLink("https://google.com")
+
+        browserScreen {
+        }.openHomeScreen {
+            verifyOpenTabsHeader()
+        }
+    }
+
+    @Test
+    fun launchPageShortcutInPrivateModeTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        setOpenLinksInPrivateOn()
+
+        addToHomeScreen {
+            addHomeScreenShortcut(defaultWebPage.url, pageShortcutName)
+        }.openHomeScreenShortcut(pageShortcutName) {
+        }.openHomeScreen {
+            verifyPrivateSessionHeader()
+        }
+    }
+
+    @Test
+    fun launchLinksInPrivateToggleOffStateDoesntChangeTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        setOpenLinksInPrivateOn()
+
+        addToHomeScreen {
+            addHomeScreenShortcut(defaultWebPage.url, pageShortcutName)
+        }.openHomeScreenShortcut(pageShortcutName) {
+        }.openHomeScreen { }
+
+        setOpenLinksInPrivateOff()
+
+        restartApp(activityTestRule)
```

Comment:
Does Espresso need to wait here? What happens if this takes a while? ('m not familiar with `finishActivity` and how it works in the context of the instrumentation runner)

Reply:
Do you think a waitForIdle would be enough?

Reply:
I missed this. I'm not too familiar with either, I think it might be worth experimenting with each.